### PR TITLE
fix: typo in license service + move to a noop

### PR DESCRIPTION
**Pull request by sean-brydon**
We were having an issue in CI where if a license key wasnt set it'd always throw an error about calcom signature. 

Now we use an NOOP to skip these license checks as we already know the value is false without a license key present

### CODE REVIEW

**Review by emrysal — DISMISSED** [review]

File: packages/features/ee/common/server/LicenseKeyService.ts [hunk 1]
```diff
@@ -23,9 +28,9 @@ class LicenseKeyService {
   }
 
   // Static async factory method
-  public static async create(): Promise<LicenseKeyService> {
+  public static async create(): Promise<ILicenseKeyService> {
     const licenseKey = await getDeploymentKey(prisma);
-    return new LicenseKeyService(licenseKey);
+    return licenseKey ? new LicenseKeyService(licenseKey) : new NoopLicenseKeyService();
```
Comment (sean-brydon) [inline packages/features/ee/common/server/LicenseKeyService.ts:33]: If theres no license key in env - we just hit the noop instance.

File: packages/features/ee/common/server/LicenseKeyService.ts [hunk 2]
```diff
@@ -94,4 +99,16 @@ class LicenseKeyService {
   }
 }
 
+export class NoopLicenseKeyService implements ILicenseKeyService {
+  async incrementUsage(_usageEvent?: UsageEvent): Promise<any> {
+    // No operation
+    return Promise.resolve();
+  }
+
+  async checkLicense(): Promise<boolean> {
```
Comment (sean-brydon) [inline packages/features/ee/common/server/LicenseKeyService.ts:108]: NOOP is only Instantiated when there is no license key - we can auto return false here.

File: packages/features/ee/common/server/LicenseKeyService.ts [hunk 3]
```diff
@@ -94,4 +99,16 @@ class LicenseKeyService {
...
+    // Always return true for NOOP
```
Comment (sean-brydon) [inline packages/features/ee/common/server/LicenseKeyService.ts:109]:
```suggestion
```

**Review by zomars — DISMISSED** [review]

**Review by Udit-takkar — COMMENTED** [review]
Unit tests and E2E tests are failing

**Review by emrysal — APPROVED** [review]

**Pull request merged**